examples/verify_tx.rs

```diff
@@ -12,18 +12,119 @@
 // If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 //
 
-//! Example: Verifying a signed transaction
+//! Example: Verifying a signed transaction.
 
 use bitcoin;
 use miniscript;
 
 use bitcoin::consensus::Decodable;
+use bitcoin::secp256k1::{self, Secp256k1};
 use bitcoin::util::sighash;
-use bitcoin::{secp256k1, TxOut}; // secp256k1 re-exported from rust-bitcoin
 use miniscript::interpreter::KeySigPair;
 use std::str::FromStr;
 
 fn main() {
+    //
+    // Setup
+    //
+
+    let tx = hard_coded_transaction();
+    let spk_input_1 = hard_coded_script_pubkey();
+
+    let interpreter = miniscript::Interpreter::from_txdata(
+        &spk_input_1,
+        &tx.input[0].script_sig,
+        &tx.input[0].witness,
+        0,
+        0,
+    )
+    .unwrap();
+
+    let desc_string = interpreter.inferred_descriptor_string();
+    println!("Descriptor: {}", desc_string);
+
+    // To do sanity checks on the transaction using the interpreter parse the
+    // descriptor with `from_str`.
+    let _ = miniscript::Descriptor::<bitcoin::PublicKey>::from_str(&desc_string)
+        .expect("sanity checks to pass");
+    // Alternately, use `inferred_descriptor` which does sanity checks for us also.
+    let _ = interpreter.inferred_descriptor().expect("same as from_str");
+
+    // Example one
+    //
+    // Learn which keys were used, not bothering to verify the signatures
+    // (trusting that if they're on the blockchain, standardness would've
+    // required they be either valid or 0-length.
+
+    println!("\n\nExample one:\n");
+
+    for elem in interpreter.iter_assume_sigs() {
+        // Don't bother checking signatures.
+        match elem.expect("no evaluation error") {
+            miniscript::interpreter::SatisfiedConstraint::PublicKey { key_sig } => {
+                let (key, sig) = key_sig
+                    .as_ecdsa()
+                    .expect("expected ecdsa sig, found schnorr sig");
+
+                println!("Signed with:\n key: {}\n sig: {}", key, sig);
+            }
+            _ => {}
+        }
+    }
+
+    // Example two
+    //
+    // Verify the signatures to ensure that invalid signatures are not treated
+    // as having participated in the script
+
+    println!("\n\nExample two:\n");
+    let secp = Secp256k1::new();
+
+    // Sometimes it is necessary to have additional information to get the
+    // `bitcoin::PublicKey` from the `MiniscriptKey` which can be supplied by
+    // the `to_pk_ctx` parameter. For example, when calculating the script
+    // pubkey of a descriptor with xpubs, the secp context and child information
+    // maybe required.
```

Comment:
Ouch, we no longer have to_pk_ctx in this codebase. This comment should have been removed. Not a part of this PR.